crates/precompile/src/bls12_381/blst.rs

```diff
@@ -115,14 +124,23 @@ fn p1_scalar_mul(p: &blst_p1_affine, scalar: &[u8]) -> blst_p1_affine {
 ///
 /// Takes a G2 point in affine form and a scalar, and returns the result
 /// of the scalar multiplication in affine form
+///
+/// Note: The scalar is expected to be in Big Endian format.
 #[inline]
-fn p2_scalar_mul(p: &blst_p2_affine, scalar: &[u8]) -> blst_p2_affine {
+fn p2_scalar_mul(p: &blst_p2_affine, scalar: &blst_scalar) -> blst_p2_affine {
     // Convert point to Jacobian coordinates
     let p_jacobian = p2_from_affine(p);
 
     let mut result = blst_p2::default();
     // SAFETY: all inputs are valid blst types
-    unsafe { blst_p2_mult(&mut result, &p_jacobian, scalar.as_ptr(), scalar.len() * 8) };
+    unsafe {
+        blst_p2_mult(
+            &mut result,
+            &p_jacobian,
+            scalar.b.as_ptr(),
+            scalar.b.len() * 8,
+        )
+    };
```

Comment:
This is now safer because by definition, we cannot pass in a scalar with more than 32 bytes

Reply:
It does remove uncertanty